colossus/src/main/scala/colossus/protocols/http/HttpRequestParser.scala

```diff
@@ -16,11 +16,13 @@ object HttpRequestParser {
   def apply(size: DataSize = DefaultMaxSize) = maxSize(size, httpRequest)
 
   protected def httpRequest: Parser[HttpRequest] = httpHead |> {head => 
-    if (head.contentLength > 0) {
-      bytes(head.contentLength) >> {body => HttpRequest(head, Some(body))}
-    } else {
-      const(HttpRequest(head, None))
-    }
+    head.singleHeader(HttpHeaders.TransferEncoding) match { 
+      case None | Some("identity") => head.contentLength match {
```

Comment:
I checked the [http spec](http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.4) and it says to ignore content length in that situation.